internal/handler/handler.go

```diff
@@ -0,0 +1,89 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/jackc/pgx/v5"
+)
+
+var ErrUserNotFound = errors.New("user not found")
+
+type TargetFunc[In Validator, Out any] func(context.Context, In) (Out, error)
+
+func Handle[In Validator, Out any](f TargetFunc[In, Out]) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var in In
+		var out Out
+
+		switch r.Method {
+		case http.MethodGet:
+			params := r.URL.Query()
+
+			jsonString := "{"
+			for key, values := range params {
+				jsonString += "\"" + key + "\":\"" + values[0] + "\","
+			}
+			jsonString = jsonString[:len(jsonString)-1] + "}"
+
+			if err := json.Unmarshal([]byte(jsonString), &in); err != nil {
+				http.Error(w, "please fill all required parameters", http.StatusBadRequest)
+				return
+			}
```

Comment:
please no. Don't do any dirty hacks.

you can use that library for example

https://github.com/monoculum/formam